featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -528,9 +530,15 @@ def _build_transform_features(self, all_features, entity, max_depth=0,
 
             for matching_input in matching_inputs:
                 if all(bf.number_output_features == 1 for bf in matching_input):
-                    new_f = GroupByTransformFeature(list(matching_input[:-1]),
-                                                    groupby=matching_input[-1],
-                                                    primitive=groupby_prim)
+                    # Need ID if it's part of the input, but not for other groupbys
+                    if id_is_input:
+                        new_f = GroupByTransformFeature(list(matching_input[:]),
+                                                        groupby=matching_input[-1],
```

Comment:
1. What if a primitive has an input_types like `[Id, Numeric]`.

2. Say an entity had two Id columns and for CUM_COUNT the matching inputs found were [id_1] and [id_2].  This code will make `CUM_COUNT(id_1) by id_1` and `CUM_COUNT(id_2) by id_2` but it should also create `CUM_COUNT(id_1) by id_2` and `CUM_COUNT(id_2) by id_1`